From 5eefc2398055dc98631e9e31c4b5554dd41d3384 Mon Sep 17 00:00:00 2001
From: Sarika Sharma <quic_sarishar@quicinc.com>
Date: Wed, 9 Oct 2024 09:18:27 +0530
Subject: [PATCH] wifi: mac80211: Fix KW issue rx->sta dereferenced without
 checking for null condition

In the function __ieee80211_rx_h_amsdu(), the rx->sta pointer is
dereferenced without verifying if it is NULL, even though it can be
NULL as indicated by a previous condition check.

Hence to fix this, check rx->sta is not NULL before proceeding with
the next condition.

patch-dependency: none

Signed-off-by: Sarika Sharma <quic_sarishar@quicinc.com>
---
 net/mac80211/rx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/mac80211/rx.c b/net/mac80211/rx.c
index 660458f..2db268b 100644
--- a/net/mac80211/rx.c
+++ b/net/mac80211/rx.c
@@ -3222,7 +3222,7 @@ __ieee80211_rx_h_amsdu(struct ieee80211_rx_data *rx, u8 data_offset)
 					  data_offset, true))
 		return RX_DROP_UNUSABLE;
 
-	if (rx->sta->amsdu_mesh_control < 0) {
+	if (rx->sta && rx->sta->amsdu_mesh_control < 0) {
 		s8 valid = -1;
 		int i;
 
-- 
2.34.1

